include/nil/blueprint/components/algebra/fields/plonk/logic_or_flag.hpp

```diff
@@ -0,0 +1,233 @@
+                    circuit<crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>> &bp,
+                    assignment<crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>> &assignment, 
+                    const typename plonk_logic_or_flag_component<BlueprintFieldType, ArithmetizationParams, WitnessesAmount>::input_type &instance_input, 
+                    const std::uint32_t start_row_index){
+
+                    std::size_t row = start_row_index;
+                    using var = typename plonk_logic_or_flag_component<BlueprintFieldType, ArithmetizationParams, WitnessesAmount>::var;
+
+                    bp.add_copy_constraint({var(component.W(0), row, false), instance_input.x});
+                    bp.add_copy_constraint({var(component.W(1), row, false), instance_input.y});
+                }
+
+                template<typename BlueprintFieldType, typename ArithmetizationParams, std::uint32_t WitnessesAmount>
+                typename plonk_logic_or_flag_component<BlueprintFieldType, ArithmetizationParams, WitnessesAmount>::result_type generate_circuit(
+                    const plonk_logic_or_flag_component<BlueprintFieldType, ArithmetizationParams, WitnessesAmount>& component,
+                    circuit<crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>> &bp,
+                    assignment<crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>> &assignment, 
+                    const typename plonk_logic_or_flag_component<BlueprintFieldType, ArithmetizationParams, WitnessesAmount>::input_type &instance_input, 
+                    const std::uint32_t start_row_index){
+
+                    auto selector_iterator = assignment.find_selector(component);
+                    std::size_t first_selector_index;
+
+                    if (selector_iterator == assignment.selectors_end()) {
+                        first_selector_index = assignment.allocate_selector(component, component.gates_amount);
+                        generate_gates(component, bp, assignment, instance_input, first_selector_index);
+                    } else {
+                        first_selector_index = selector_iterator->second;
+                    }
+                    assignment.enable_selector(first_selector_index, start_row_index + component.rows_amount == 3 ? 1 : 0);
+                    if(WitnessesAmount == 2){
+                        assignment.enable_selector(first_selector_index+1, start_row_index + 2);
+                    }
+
+                    generate_copy_constraints(component, bp, assignment, instance_input, start_row_index);
+
+                    return typename plonk_logic_or_flag_component<BlueprintFieldType, ArithmetizationParams, WitnessesAmount>::result_type(component, start_row_index);
+                }
+        }
+    }
+}
+
+#endif CRYPTO3_BLUEPRINT_COMPONENTS_PLONK_FIELDS_logic_OR_FLAG_HPP
```

Comment:
This should be `#endif    // CRYPTO3_BLUEPRINT_COMPONENTS_PLONK_FIELDS_LOGIC_OR_FLAG_HPP`, otherwise compiler shows a warning.